demos/device_defender_for_aws/defender_demo.c

```diff
@@ -0,0 +1,649 @@
+
+    if( status == true )
+    {
+        status = UnsubscribeFromTopic( pxMqttContext,
+                                       DEFENDER_API_JSON_REJECTED( THING_NAME ),
+                                       DEFENDER_API_LENGTH_JSON_REJECTED( THING_NAME_LENGTH ) );
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static int generateDeviceMetricsReport( uint32_t * pOutReportLength )
+{
+    int status = EXIT_FAILURE;
+    ReportBuilderStatus_t reportBuilderStatus;
+
+    /* Generate the metrics report in the format expected by the AWS IoT Device
+     * Defender Service. */
+    reportBuilderStatus = GenerateJsonReport( &( deviceMetricsJsonReport[ 0 ] ),
+                                              DEVICE_METRICS_REPORT_BUFFER_SIZE,
+                                              &( deviceMetrics ),
+                                              DEVICE_METRICS_REPORT_MAJOR_VERSION,
+                                              DEVICE_METRICS_REPORT_MINOR_VERSION,
+                                              reportId,
+                                              pOutReportLength );
+
+    if( reportBuilderStatus != ReportBuilderSuccess )
+    {
+        IotLogError( "GenerateJsonReport failed. Status: %d.",
+                    reportBuilderStatus );
+    }
+    else
+    {
+        IotLogInfo( "Generated Report: %.*s.",                    //MingL: Used to be Debug
+                    *pOutReportLength,
+                    &( deviceMetricsJsonReport[ 0 ] ) );
+        status = EXIT_SUCCESS;
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The function that runs the Defender demo, called by the demo runner.
+ *
+ * @param[in] awsIotMqttMode Ignored for the Defender demo.
+ * @param[in] pIdentifier NULL-terminated Defender Thing Name.
+ * @param[in] pNetworkServerInfo Passed to the MQTT connect function when
+ * establishing the MQTT connection for Defender.
+ * @param[in] pNetworkCredentialInfo Passed to the MQTT connect function when
+ * establishing the MQTT connection for Defender.
+ * @param[in] pNetworkInterface The network interface to use for this demo.
+ *
+ * @return `EXIT_SUCCESS` if the demo completes successfully; `EXIT_FAILURE` otherwise.
+ */
+
+int RunDeviceDefenderDemo( bool awsIotMqttMode,
+                     const char * pIdentifier,
+                     void * pNetworkServerInfo,
+                     void * pNetworkCredentialInfo,
+                     const void * pNetworkInterface )
+{
+    int exitStatus = EXIT_FAILURE;
+    uint32_t reportLength = 0, i, mqttSessionEstablished = 0;
+
+
+    /* Start with report not received. */
+    reportStatus = ReportStatusNotReceived;
+
+    /* Set a report Id to be used. */
+    reportId = 1;
+
+    IotLogInfo( "Establishing MQTT session..." );
+    exitStatus = EstablishMqttSession( &xMqttContext,
```

Comment:
This function returns pdPASS or pdFAIL please update for this status

Reply:
Made a commit to update and align the function commit type with demo helper.
e9d96e2